server/restafary.js

```diff
@@ -203,7 +216,13 @@ async function onGet(p, callback) {
             params.name = path;
         
         params.gzip = false;
-        ponse.sendFile(params);
+        
+        // Only process the archive listing if we are not given a query command
+        if (!p.request.originalUrl.includes('?') && archive.isArchive(path)) {
+            await sendArchiveListing(path, params);
```

Comment:
I think you can simplify check to `if (p.query)`

Reply:
p.query always contains either the api path or the query command, so, I can't use a check on p.query.